### PR TITLE
Forbid optimized plans to be analyzed [HZ-3838]

### PR DESCRIPTION
**Pull request by Fly-Style**
For `IMapByKey` plans - we return an error with the suggestion to run the usual query without ANALYZE

### CODE REVIEW

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java [hunk 1]
```diff
@@ -1030,6 +1035,18 @@ public static Map<String, List<Map<String, Expression<?>>>> partitionStrategyCan
...
     }
 
+    /**
+     * Check should be used during the optimized IMapByKey plan construction.
+     * It throws {@link QueryException} of query is analyzed for optimized IMapByKey plans.
+     * @param isAnalyzed is query analyzed
+     */
+    static void checkIMapByKeyPlanIsAnalyzed(boolean isAnalyzed) {
+        if (isAnalyzed) {
+            throw QueryException.error("ANALYZE statement is not applicable for key-based optimized IMap access plans." +
```
Comment (k-jamroz) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java:1045]: I would work on a better message. The user is not required to know what plan _will be chosen by the optimizer_ and the SQL syntax is ordinary. Something like "This query uses key-based optimized IMap access plan. ANALYZE is unable to produce meaningful execution statistics for it."

Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java]: https://github.com/hazelcast/hazelcast/pull/26002/commits

**Review by k-jamroz — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java [hunk 2]
```diff
@@ -649,6 +651,7 @@ private SqlPlanImpl toPlan(
             );
         } else if (physicalRel instanceof SinkMapPhysicalRel) {
             assert !isCreateJob;
+            checkIMapByKeyPlanIsAnalyzed(analyze);
             SinkMapPhysicalRel sink = (SinkMapPhysicalRel) physicalRel;
```
Comment (shultseva) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java:652]: which test checks this part?

Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java:652]: https://github.com/hazelcast/hazelcast/pull/26002/commits/a3b54c6c0f17406a240288c3edbcb514509f17fa

**Review by shultseva — APPROVED** [review]

**Pull request merged**